### PR TITLE
added CoreData as database , and used only on homepage

### LINKED ISSUE #146: Offline Support using Core Data
**Is your feature request related to a problem? Please describe.**
Currently the offline support in Mentorship iOS app is very limited (only for user profile). This makes it difficult for a user to be able to access their data at all the time, as it is not possible with wifi/data. Also, since the same data is fetched from the server again and again, user's data also gets wasted which can be conserved with offline support.

**Describe the solution you'd like**
Core Data framework must be used to enable offline support in the app. The boilerplate code for core data is present in the project. We have implemented repository pattern in the app (as a Service layer) which allows for which data sources to be used for the data model. The offline data source should ideally follow this paradigm.

**Describe alternatives you've considered**
The profile uses UserDefaults for offline support. However, I feel that core data would be the better option for the whole app.

**Additional context**
We can be flexible on the scope of this issue and only implement offline support for the home screen for now (instead of the whole app).
Note: This issue may also require a little refactoring during the process.

### PR DESCRIPTION
**Pull request by paraschhugani**
closes #146 
I have added Coredata and only used in the home screen showing the Name of user,, 
you may see many files changed because i installed the pods and the newer version of pods was made in-app.
You may ignore that files.

https://user-images.githubusercontent.com/51089028/103199853-df074780-4911-11eb-84f5-54b184396227.mov

### CODE REVIEW

File: mentorship ios/Service/Networking/LoginAPI.swift [hunk 1]
```diff
@@ -32,6 +34,7 @@ class LoginAPI: LoginService {
                         loginResponseData.message = "Failed to save access token"
                     }
                 }
+//                CoreUserProfile().test(token: String(response.accessToken!))
```
Comment (vatsalkul) [inline mentorship ios/Service/Networking/LoginAPI.swift:37]: remove this

File: mentorship ios/Service/Networking/LoginAPI.swift [hunk 2]
```diff
@@ -23,6 +23,8 @@ class LoginAPI: LoginService {
             .sink { response in
                 var loginResponseData = LoginModel.LoginResponseData(message: response.message)
                 // if login successful, store access token in keychain
+                print("here testting the decdoed respose\(response)")
+//                CoreUserProfile().saveCore(token: response.accessToken!)
```
Comment (vatsalkul) [inline mentorship ios/Service/Networking/LoginAPI.swift:27]: remove this

**Review by vatsalkul — CHANGES_REQUESTED** [review]
Hey @paraschhugani I have looked into the code and I have few queries.
1. Travis CI build is failing and that is because you have used loadCoreData() in closure without using a explicit self. Please take a look into that
2. You made 2 loadCoreData function, Why? Can you try to reuse the same function. You can return the optional HomeUserName from loadCoreData function. (It's your choice)
3. Please remove any commented code which is not useful.
4. Try not to use print statement. You can use NSLog for important informations.
5. Ignore pods as well

File: mentorship ios/Service/Networking/LoginAPI.swift (hunk 2 above)
Comment (paraschhugani) [reply mentorship ios/Service/Networking/LoginAPI.swift:27]: thanks for review, 
I will do the changes,
 I have used 2 loadCoredata() function so in future it is easy to customise for different data tables.
can you please say how will I ignore pods, the build is crashing without installing them

**Pull request closed without merging**